pkg/fanal/analyzer/language/dotnet/nuget/nuget.go

```diff
@@ -2,59 +2,112 @@ package nuget
 
 import (
 	"context"
+	"errors"
+	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
+	"sort"
 
 	"golang.org/x/exp/slices"
-
 	"golang.org/x/xerrors"
 
 	"github.com/aquasecurity/go-dep-parser/pkg/nuget/config"
 	"github.com/aquasecurity/go-dep-parser/pkg/nuget/lock"
+	godeptypes "github.com/aquasecurity/go-dep-parser/pkg/types"
 	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
 	"github.com/aquasecurity/trivy/pkg/fanal/analyzer/language"
 	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
 )
 
 func init() {
-	analyzer.RegisterAnalyzer(&nugetLibraryAnalyzer{})
+	analyzer.RegisterPostAnalyzer(types.NuGet, newNugetLibraryAnalyzer)
 }
 
 const (
-	version    = 2
+	version    = 3
 	lockFile   = types.NuGetPkgsLock
 	configFile = types.NuGetPkgsConfig
 )
 
 var requiredFiles = []string{lockFile, configFile}
 
-type nugetLibraryAnalyzer struct{}
+type nugetLibraryAnalyzer struct {
+	lockParser    godeptypes.Parser
+	configParser  godeptypes.Parser
+	licenseParser nuspecParser
+}
 
-func (a nugetLibraryAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
-	// Set the default parser
-	parser := lock.NewParser()
+func newNugetLibraryAnalyzer(_ analyzer.AnalyzerOptions) (analyzer.PostAnalyzer, error) {
+	return &nugetLibraryAnalyzer{
+		lockParser:    lock.NewParser(),
+		configParser:  config.NewParser(),
+		licenseParser: newNuspecParser(),
+	}, nil
+}
 
-	targetFile := filepath.Base(input.FilePath)
-	if targetFile == configFile {
-		parser = config.NewParser()
+func (a *nugetLibraryAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysisInput) (*analyzer.AnalysisResult, error) {
+	var apps []types.Application
+	foundLicenses := make(map[string][]string)
+
+	// We saved only config and lock files in the FS,
+	// so we need to parse all saved files
+	required := func(path string, d fs.DirEntry) bool {
+		return true
 	}
 
-	res, err := language.Analyze(types.NuGet, input.FilePath, input.Content, parser)
+	err := fsutils.WalkDir(input.FS, ".", required, func(path string, d fs.DirEntry, r io.Reader) error {
+		// Set the default parser
+		parser := a.lockParser
+
+		targetFile := filepath.Base(path)
+		if targetFile == configFile {
+			parser = a.configParser
+		}
+
+		app, err := language.Parse(types.NuGet, path, r, parser)
+		if err != nil {
+			return xerrors.Errorf("NuGet parse error: %w", err)
+		}
+
+		for i, lib := range app.Libraries {
+			license, ok := foundLicenses[lib.ID]
+			if !ok {
+				license, err = a.licenseParser.findLicense(lib.Name, lib.Version)
+				if err != nil {
+					if !errors.Is(err, fs.ErrNotExist) {
+						return xerrors.Errorf("license find error: %w", err)
+					}
+				}
```

Comment:
nit: We might want to aggregate the statements.

```suggestion
				if err != nil && !errors.Is(err, fs.ErrNotExist) {
					return xerrors.Errorf("license find error: %w", err)
				}
```

Reply:
Right! Thanks! Updated in https://github.com/aquasecurity/trivy/pull/5217/commits/1d0fed1336f1501c5d1f3d1b453be7fc47fa1ba9.